identity-apps-core/apps/authentication-portal/src/main/webapp/templates/internalWait.jsp

```diff
@@ -0,0 +1,290 @@
+<%--
+ ~ Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ ~
+ ~ This software is the property of WSO2 LLC. and its suppliers, if any.
+ ~ Dissemination of any information or reproduction of any material contained
+ ~ herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ ~ You may not alter or remove any copyright or other notice from copies of this content.
+--%>
+
+<%--  
+    This template is used to keep an user waiting using the adaptive auth dynamic prompt capability.
+    An example prompt usage in adaptive script is shown below. The parameters are optional and if not provided,
+    the default values will be used.
+
+    prompt("internalWait", {
+        "type": "<WAITING_METHOD>",
+        "data": {
+            "greeting": "<GREETING_TEXT>",
+            "message": "<WAITING_MESSAGE>",
+            "pollingEndpoint": "<POLLING_ENDPOINT>",
+            "requestMethod": "<POLLING_REQUEST_METHOD>",
+            "contentType": "<POLLING_REQUEST_CONTENT_TYPE>",
+            "requestData": "<POLLING_REQUEST_DATA>",
+            "pollingInterval": "<POLLING_INTERVAL_SECONDS>",
+            "timeout": "<TIMEOUT_SECONDS>"
+        }
+    }, {
+        onSuccess: function(context) {
+            Log.info("Successfully redirected back from waiting page.");
+        },
+        onFail: function(context, data) {
+            Log.info("Error occurred during the prompt.");
+        }
+    });
+--%>
+
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8" %>
+<%@ page import="java.io.File" %>
+<%@ page import="java.util.Arrays" %>
+<%@ page import="java.util.HashMap" %>
+<%@ page import="java.util.List" %>
+<%@ page import="java.util.Map" %>
+<%@ page import="org.apache.commons.lang.StringUtils" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.AuthenticationEndpointUtil" %>
+
+<jsp:directive.include file="../includes/init-url.jsp"/>
+
+<%-- Localization --%>
+<jsp:directive.include file="../includes/localize.jsp" />
+
+<%-- Branding Preferences --%>
+<jsp:directive.include file="../includes/branding-preferences.jsp"/>
+
+<%
+    // Read promptId from request.
+    String promptId = (String) request.getAttribute("promptId");
```

Comment:
Could you check whether we need to sanitize the request params here?

Reply:
This page gets loaded into [dynamic_prompt.jsp](https://github.com/wso2/identity-apps/blob/368db4133456ba6015b1bb812715350c8d26834e/identity-apps-core/apps/authentication-portal/src/main/webapp/dynamic_prompt.jsp) just like other templates available in https://github.com/wso2/identity-apps/tree/368db4133456ba6015b1bb812715350c8d26834e/identity-apps-core/apps/authentication-portal/src/main/webapp/templates. Hence do we need sanitization here?

Reply:
Its better to add sanitisation in such places, because in future it will help us to avoid any regressions.

Reply:
Addressed.

Reply:
Addressed